external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -218,7 +223,11 @@ public void nextTuple() {
             }
 
             if (poll()) {
-                setWaitingToEmit(pollKafkaBroker());
+                try {
+                    setWaitingToEmit(pollKafkaBroker());
+                } catch (KafkaException e) {
```

Comment:
I think you should catch https://kafka.apache.org/0100/javadoc/org/apache/kafka/common/errors/RetriableException.html instead, it seems to cover all the exceptions that are expected to be transient (leader change for example, through InvalidMetadataException). Catching KafkaException will break interrupt handling (see https://github.com/apache/storm/pull/1821) because InterruptException is an instance of KafkaException.